### PR TITLE
Migrate admin review page to Django.

### PR DESCRIPTION
**Pull request by nworden**

### CODE REVIEW

**Comment by nworden** [comment]
Sent to Christophe.

File: app/views/base.py [hunk 1]
```diff
@@ -251,11 +254,23 @@ def build_absolute_path(self, path=None, repo=None):
...
-            return '/%s%s' % (site_settings.OPTIONAL_PATH_PREFIX, path)
+            res = '/%s%s' % (site_settings.OPTIONAL_PATH_PREFIX, path)
         else:
-            return path
-
-    def build_absolute_uri(self, path=None, repo=None):
+            res = path
+        if params:
+            url_parts = list(urlparse.urlparse(res))
+            url_params = dict(urlparse.parse_qsl(url_parts[4]))
```
Comment (christopherestif) [inline app/views/base.py]: is it worth documenting the number 4?

File: app/views/admin/review.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+                # Copy in the fields of the associated Person.
+                for name in person.properties():
+                    setattr(note, 'person_' + name, getattr(person, name))
+
+                # Get the statuses of the other notes on this Person.
+                status_codes = ''
+                for other_note in person.get_notes():
+                    code = AdminReviewView._STATUS_CODES[other_note.status]
+                    if other_note.note_record_id == note.note_record_id:
+                        code = code.upper()
```
[the remark this answers is not in the document]
Comment (nworden) [reply app/views/admin/review.py]: This code was taken directly from the original webapp2 code. I think what's happening is it's showing each of the statuses for a person as a single letter (for compactness), and the status from the current note is uppercased to show it's the current note's.

File: app/views/admin/review.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+                    '/admin/review', self.env.repo,
+                    params=[
+                        ('source', option),
+                        ('status', current_selected_status),
+                    ])
+                source_options_nav.append((option, option_url))
+
+        query = model.Note.all_in_repo(self.env.repo)
+        query = query.filter('reviewed =', False)
+        query = query.filter('hidden =', False)
```
[the remark this answers is not in the document]
Comment (nworden) [reply app/views/admin/review.py]: It was originally, but the formatting gets awkward. The original was something like:

```python
query = model.Note.all_in_repo(
    self.env.repo).filter(
    'reviewed =', False).filter(
    'hidden =', False)
```
So I just split it up. I've combined the two filter lines now though.

File: app/views/base.py (hunk 1 above)
Comment (nworden) [reply app/views/base.py]: IMO the context of assigning it to the `url_params` variable is enough.

**Pull request merged**